supervisor/helpers.py

```diff
@@ -47,20 +49,52 @@ def message(self):
             msg = "exit status %s" % (self.returncode,)
         return msg
 
-    def send_os_signal(self, sig, proc_name):
-        """Send signal by GenerateConsoleCtrlEvent"""
-        status = ctypes.windll.kernel32.GenerateConsoleCtrlEvent(sig, self.pid)
+    def check_win32api_status(self, status):
         if status == 0:
             status = win32api.FormatMessage(win32api.GetLastError())
             status = as_string(status, errors='ignore')
             status = status.strip('\r\n')
+        return status
+
+    def send_os_signal(self, sig, proc_name):
+        """Send signal by GenerateConsoleCtrlEvent"""
+        status = ctypes.windll.kernel32.GenerateConsoleCtrlEvent(sig, self.pid)
+        status = self.check_win32api_status(status)
         return "signal: %s (status %s)" % (proc_name, status)
 
+    def send_win_signal(self, sig, proc_name):
+        window = []
+
+        # callback for handling window search
+        def check_window(hwnd, window):
+            _, pid = win32process.GetWindowThreadProcessId(hwnd)
+            if pid == self.pid:
+                window.append(hwnd)
+                return False
+            return True
+
+        # find window
+        try:
+            win32gui.EnumWindows(check_window, window)
+        except pywintypes.error:
+            # Exception is raised by terminating search early (if a window is found)
+            # Ignore it!
+            pass
+
+        if window:
+            status = win32api.PostMessage(window[0], sig, None, None)
+            status = self.check_win32api_status(status)
+            return "win signal: %s (status %s)" % (proc_name, status)
+        else:
+            return "Failed to find window to send signal to."
```

Comment:
Can you name the process and the reason for the failure as in this example below ?
`"win signal: %s (status  failed to find window to send signal)" % (proc_name,)`

Reply:
Done